test/ruleset/rules/asyncapi-document-resolved.spec.ts

```diff
@@ -309,4 +309,89 @@ testRule('asyncapi-document-resolved', [
     },
     errors: [],
   },
+
+  {
+    name: 'valid case (3.0.0 version)',
+    document: {
+      asyncapi: '3.0.0',
+      info: {
+        title: 'Signup service example (internal)',
+        version: '0.1.0',
+      },
+      channels: {
+        'user/signedup': {
+          address: 'user/signedup',
+          messages: {
+            'subscribe.message': {
+              payload: {}
+            }
+          }
+        }
+      },
+      operations: {
+        'user/signedup.subscribe': {
+          action: 'send',
+          channel: {
+            address: 'user/signedup',
+            messages: {
+              'subscribe.message': {
+                payload: {}
+              }
+            }
+          },
+          messages: [
+            {
+              payload: {}
+            }
+          ]
+        }
+      },
+    },
+    errors: [],
+  },
+
+  {
+    name: 'invalid case for 3.X.X (info.version property is missing)',
+    document: {
+      asyncapi: '3.0.0',
+      info: {
+        title: 'Valid AsyncApi document',
+      },
+    },
+    errors: [
+      {
+        message: '"info" property must have required property "version"',
+        severity: DiagnosticSeverity.Error,
+      },
+    ],
+  },
+
+  {
+    name: 'valid case for 3.X.X (case when other $ref errors should also occur but we filter them out)',
```

Comment:
Are these names on cases still valid?

Reply:
Spectral works in two modes - with schema resolved and unresolved (with untouched refs). Even if we wrote a ref in a place then Spectral would not throw an error (because it operates on resolved schema). This is more of a test for checking if rule isn't "broking" resolved/unresolved spectral behaviour, just in case for us :)

Reply:
If you want you can change the name of the test because it should have a different name, I can't remember why I called it that.